### PR TITLE
[bootstrap] Add swift-argument-parser as a bootstrapped swift-driver dependency

### PR DESCRIPTION
**Pull request by owenv**
Only the last commit is new, the rest is from #2736

### CODE REVIEW

File: Utilities/bootstrap [hunk 1]
```diff
@@ -289,6 +291,9 @@ def test(args):
     # Test Yams.
     call_swiftpm(args, cmd, args.yams_source_dir)
 
+    # Test swift-argument-parser.
```
Comment (DougGregor) [inline Utilities/bootstrap:294]: I backed away from testing Yams and SwiftDriver, and I think this should be removed as well. Let's just test SwiftPM here.

**Comment by owenv** [comment]
Marking this as a draft until I have a chance to make the update-checkout changes

File: Sources/Build/CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +20,7 @@ target_link_libraries(Build PUBLIC
   SPMBuildCore
   CYaml
   Yams
+  ArgumentParser
```
Comment (owenv) [inline Sources/Build/CMakeLists.txt:23]: This isn't strictly necessary yet, but I think it'll allow incremental adoption of ArgumentParser in the driver library

**Comment by owenv** [comment]
https://github.com/apple/swift-driver/pull/92
https://github.com/apple/swift/pull/31790

@swift-ci please test

**Comment by owenv** [comment]
There might be some remaining build issues on Linux to resolve here, but let's see how it works with an updated ArgumentParser

apple/swift-driver#92
@swift-ci please smoke test

**Comment by owenv** [comment]
apple/swift-driver#92
@swift-ci please smoke test

**Comment by owenv** [comment]
apple/swift-driver#92
@swift-ci please smoke test

**Comment by owenv** [comment]
apple/swift-driver#92
@swift-ci please smoke test

**Comment by DougGregor** [comment]
FYI, also learned about Docker as another place to update: https://github.com/apple/swift-package-manager/pull/2767

**Comment by tomerd** [comment]
@DougGregor this seems to be still draft. let us know when you need a review

**Comment by DougGregor** [comment]
We should go ahead and do this when it's ready. Are you stuck on something, @owenv ?

**Comment by owenv** [comment]
@DougGregor I think we'll want to coordinate this with @natecook1000's changes in https://github.com/apple/swift-package-manager/pull/2653 so that CI & update-checkout can be configured to pull a single version of ArgumentParser for use in both swift-driver and SPM. I'll try and test this with swift-argument-parser 0.3 sometime in the next few days

**Comment by owenv** [comment]
https://github.com/apple/swift-driver/pull/92
@swift-ci smoke test

**Comment by owenv** [comment]
https://github.com/apple/swift-driver/pull/92
@swift-ci smoke test

**Comment by owenv** [comment]
https://github.com/apple/swift-driver/pull/92
@swift-ci test

**Comment by abertelrud** [comment]
@owenv Please merge when ready (or I can do it, if it can go in at any time).

**Comment by owenv** [comment]
@abertelrud thanks for the reminder! I just realized I forgot to request a review on the corresponding swift-driver PR (https://github.com/apple/swift-driver/pull/92). Once that's approved, I might need help with the cross-repository merge (I don't have write access to this repository)

**Comment by abertelrud** [comment]
@owenv No problem, just let me know when this is ready to merge.  Thanks!

**Comment by owenv** [comment]
@abertelrud it looks like everything is approved so this should be ready to merge now. Thanks for your help with this!

**Comment by DougGregor** [comment]
Let's do it!

**Pull request merged**